test/unit/app-check/token-generator.spec.ts

```diff
@@ -122,11 +122,49 @@ describe('AppCheckTokenGenerator', () => {
       }).to.throw(FirebaseAppCheckError).with.property('code', 'app-check/invalid-argument');
     });
 
-    it('should be fulfilled with a Firebase Custom JWT', () => {
+    const invalidOptions = [null, NaN, 0, 1, true, false, [], _.noop];
+    invalidOptions.forEach((invalidOption) => {
+      it('should throw given an invalid options: ' + JSON.stringify(invalidOption), () => {
+        expect(() => {
+          tokenGenerator.createCustomToken(APP_ID, invalidOption as any);
+        }).to.throw(FirebaseAppCheckError).with.property('message', 'AppCheckTokenOptions must be a non-null object.');
+      });
+    });
+
+    const invalidTtls = [null, NaN, '0', 'abc', '', -100, -1, true, false, [], {}, { a: 1 }, _.noop];
+    invalidTtls.forEach((invalidTtl) => {
+      it('should throw given an options object with invalid ttl: ' + JSON.stringify(invalidTtl), () => {
+        expect(() => {
+          tokenGenerator.createCustomToken(APP_ID, { ttlMillis: invalidTtl as any });
+        }).to.throw(FirebaseAppCheckError).with.property('message',
+          'ttlMillis must be a non-negative duration in milliseconds.');
+      });
+    });
+
+    const THIRTY_MIN_IN_MS = 1800000;
+    const SEVEN_DAYS_IN_MS = 604800000;
+    [0, 10, THIRTY_MIN_IN_MS - 1, SEVEN_DAYS_IN_MS + 1, SEVEN_DAYS_IN_MS * 2].forEach((ttlMillis) => {
+      it('should throw given options with ttl < 30 minutes or ttl > 7 days:' + JSON.stringify(ttlMillis), () => {
+        expect(() => {
+          tokenGenerator.createCustomToken(APP_ID, { ttlMillis });
+        }).to.throw(FirebaseAppCheckError).with.property(
+          'message', 'ttlMillis must be a duration in milliseconds between 30 minutes and 7 days (inclusive).');
+      });
+    });
+
+    it('should be fulfilled with a Firebase Custom JWT with only an APP ID', () => {
       return tokenGenerator.createCustomToken(APP_ID)
         .should.eventually.be.a('string').and.not.be.empty;
     });
 
+    [THIRTY_MIN_IN_MS, THIRTY_MIN_IN_MS + 1, SEVEN_DAYS_IN_MS / 2, SEVEN_DAYS_IN_MS - 1, SEVEN_DAYS_IN_MS]
+      .forEach((ttlMillis) => {
+        it('should be fulfilled with a Firebase Custom JWT with a valid custom ttl' + JSON.stringify(ttlMillis), () => {
+          return tokenGenerator.createCustomToken(APP_ID, { ttlMillis })
```

Comment:
We should probably decode the token and verify the expected TTL is set.

Reply:
Good call. Thanks!